content-src/selectors/selectors.js

```diff
@@ -7,26 +7,59 @@ const urlParse = require("url-parse");
 const SPOTLIGHT_LENGTH = module.exports.SPOTLIGHT_LENGTH = 3;
 const TOP_SITES_LENGTH = module.exports.TOP_SITES_LENGTH = 6;
 const BACKGROUND_FADE = 0.5;
-const DEFAULT_FAVICON_BG_COLOR = [255, 255, 255];
+const DEFAULT_FAVICON_BG_COLOR = [150, 150, 150];
 
 module.exports.justDispatch = (() => ({}));
 
+function getBackgroundRGB(site) {
+  if (site.favicon_colors && site.favicon_colors[0] && site.favicon_colors[0].color) {
+    return site.favicon_colors[0].color;
+  }
+
+  const favicon = site.favicon_url || site.favicon;
+  const parsedUrl = site.parsedUrl || urlParse(site.url || "") ;
+  const label = prettyUrl(parsedUrl.hostname);
+  let backgroundRGB = favicon ? DEFAULT_FAVICON_BG_COLOR : getRandomColor(label);
+  try { backgroundRGB = site.favicon_colors[0].color || backgroundRGB; } catch (e) {/**/}
```

Comment:
This line confuses me. Is it needed? We check for `site.favicon_colors[0].color` at the top and return early if it exists.